repos/fdbt-site/src/pages/defineCapPricingPerDistance.tsx

```diff
@@ -0,0 +1,207 @@
+                                        const items = [...capPricingPerDistanceData.capPricing];
+                                        let item = { ...capPricingPerDistanceData.capPricing[numberOfCap - 1] };
+                                        item.distanceTo = '';
+                                        items[numberOfCap - 1] = item;
+                                        setCapPricingPerDistanceData({
+                                            ...capPricingPerDistanceData,
+                                            capPricing: items,
+                                        });
+                                        item = { ...capPricingPerDistanceData.capPricing[numberOfCap] };
+                                        item.distanceTo = 'Max';
+                                        items[numberOfCap] = item;
+                                        setCapPricingPerDistanceData({
+                                            ...capPricingPerDistanceData,
+                                            capPricing: items,
+                                        });
+                                    }}
+                                >
+                                    Add another row
+                                </button>
+                            ) : (
+                                ''
+                            )}
+
+                            {numberOfCap > 1 ? (
+                                <button
+                                    id="remove-button"
+                                    type="button"
+                                    className="govuk-button govuk-button--secondary govuk-!-margin-left-3 govuk-!-margin-bottom-3"
+                                    onClick={(): void => {
+                                        setNumberOfCaps(numberOfCap - 1);
+                                    }}
+                                >
+                                    Remove last row
+                                </button>
+                            ) : (
+                                ''
+                            )}
+                        </div>
+                        <input
+                            type="submit"
+                            value="Continue"
+                            id="continue-button"
+                            className="govuk-button govuk-!-margin-left-3"
+                        />
+                    </div>
+                </>
+            </CsrfForm>
+        </FullColumnLayout>
+    );
+};
+
+export const getServerSideProps = (ctx: NextPageContextWithSession): { props: DefineCapPricingPerDistanceProps } => {
+    const csrfToken = getCsrfToken(ctx);
+    const capPricePerDistances: DistanceCap | WithErrors<DistanceCap> | undefined = getSessionAttribute(
+        ctx.req,
+        CAP_PRICING_PER_DISTANCE_ATTRIBUTE,
+    );
+    let errors: ErrorInfo[] = [];
+    if (capPricePerDistances && 'errors' in capPricePerDistances) {
```

Comment:
we have a typeguard function called isWithErrors I believe, that we can use here